Compiletest

Redecorating opened the pull request:
Test applying the patches and compiling the kernel with only modules for t2 hardware enabled.

Use a file called `version` to determine which kernel version to apply the patches to.

In future I'd like to have it create a tag for the version if:

a) no such tag already exists
b) the compiletest doesn't have errors
c) it's on the main branch

It might then be able to trigger CI's for our different T2 distros

we might want to make the version have a -x at the end e.g. 6.1.2-1 so
we can release fixes between versions

Redecorating committed:
Update extra_config

Redecorating committed:
Create compile_test.yml

not sure how best to handle the version picking, maybe we add a file containing the current version's name?

Redecorating committed:
Update compile_test.yml

Redecorating committed:
Update compile_test.yml

https://github.com/t2linux/T2-Ubuntu-Kernel/blob/5c3f3708fe7c9bf991a2a9b5521a4218467f9a12/build.sh#L56 very helpful :)

Redecorating committed:
Update compile_test.yml

Redecorating committed:
Update extra_config

I'm sure I've missed lots of things but this will do for now

Redecorating committed:
compiletest: get version from file

thus you can just edit that file to update the new version.

later we can probably make it create a tag with that version if:

a) no such tag already exists
b) the compiletest doesn't have errors
c) it's on the main branch

we might want to make the version have a -x at the end e.g. 6.1.2-1 so
we can release fixes between versions

AdityaGarg8 commented:
We probably can exclude bcm5974 and applesmc from checkpatch since they shall not be sent upstream

AdityaGarg8 commented on .github/workflows/compile_test.yml:
```diff
@@ -0,0 +1,59 @@
+      run: |
+        KVER=$(cat version)
+        cd linux-$KVER
+        # Skip apple-bce because it needs a lot of formatting changes, and
```
We probably can exclude bcm5974 and applesmc from checkpatch since they shall not be sent upstream

Redecorating committed:
don't checkpatch applesmc and bcm5974

Redecorating committed:
compiletest: allow more variables in version file

in future this should let us put the release number in the same file (in
case we need to push another release before the next stable linux
version comes out)

AdityaGarg8 commented:
Btw over here do we intend to update the kernel version in this repo and push the same to all the kernel repos?

AdityaGarg8 commented:
Use an action similar to https://github.com/marketplace/actions/push-directory-to-another-repository

Redecorating commented:
> Btw over here do we intend to update the kernel version in this repo and push the same to all the kernel repos?

we could do that in the future. For arch at least i've been waiting until after the official arch version of a kernel is out before releasing a t2 arch kernel, so that probably would change if we automate it as you've described.

AdityaGarg8 commented:
> we could do that in the future. For arch at least i've been waiting until after the official arch version of a kernel is out before releasing a t2 arch kernel, so that probably would change if we automate it as you've described.

Fair enough. I've been using a cron job which watches kernel.org and updates the repo automatically for Ubuntu.

Redecorating committed:
Update compile_test.yml

avoids having it run the  ci twice each time something is pushed to a branch that has a pr
